### ISSUE TITLE
I am sure I've made a mistake, but confused about the behavior of ``attention_mask'' of the roformer. 

### ISSUE THREAD

**Issue by aalennku** [huggingface/transformers#33318]
### System Info

I am confused about the behavior of ``attention_mask'' of the roformer. 
As the document in https://huggingface.co/docs/transformers/en/model_doc/roformer#transformers.RoFormerModel.forward.attention_mask 
said 0 for masked and 1 for not masked. 

but the source code in 

src/transformers/models/roformer/modeling_roformer.py
Line 282 : attention_scores = attention_scores + attention_mask

uses the old fashioned transformer style 0 for unmasked and -inf for masked? 

@ArthurZucker

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [x] My own task or dataset (give details below)

### Reproduction

 attention_scores = attention_scores + attention_mask

### Expected behavior

 attention_scores = attention_scores * attention_mask

**Comment by ArthurZucker** [comment]
Hey! The attention mask is changed (inverted and extended to 4D) here: https://github.com/huggingface/transformers/blob/0c8021c5d566bcc6897ba3712412414ab47f5e44/src/transformers/models/roformer/modeling_roformer.py#L878-L880

**Comment by aalennku** [comment]
> nded_attention_mask(attention_mask, input_shape)

Thank you!
